hastory-server/src/Data/Hastory/Server/Utils.hs

```diff
@@ -2,13 +2,33 @@
 
 module Data.Hastory.Server.Utils where
 
+import Control.Monad.Except
+import Control.Monad.Reader as X
 import Crypto.Hash (Digest, SHA256(..), hashWith)
 import qualified Data.ByteString as B
-import Data.Hastory.Types (Entry(..), ServerEntry(..), SyncRequest(..))
+import Data.Hastory.Types
+import Data.Pool (Pool)
 import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
 import Data.Time.Format (defaultTimeLocale, formatTime, iso8601DateFormat)
+import Database.Persist.Sql
 import Path (fromAbsDir)
+import Servant.Auth.Server
+import Servant.Server
+
+data ServerSettings =
+  ServerSettings
+    { _ssDbPool :: Pool SqlBackend
+      -- ^ A pool of database connections.
+        --
+        -- Curently, the database file is located at "~/hastory-data/hastory.db"
```

Comment:
It's nice that you comment the code, but this part of the code couldn't care less where the db file comes from.
It's the responsibility of the `OptParse` mechanism to take care of that.

Reply:
👍